libraries/holochain-service-client/src/main/java/org/holochain/androidserviceruntime/holochain_service_client/HolochainServiceClient.kt

```diff
@@ -31,13 +32,48 @@ class HolochainServiceClient(
         }
       }
 
+  /// Entire process to setup an app
+  ///
+  /// Connect to service, install app, enable app, ensure app websocket
+  suspend fun setupApp(
+      installAppPayload: InstallAppPayloadFfi,
+      enableAfterInstall: Boolean
+  ): AppAuthFfi {
+    Log.d(TAG, "setupApp")
+    this.connect()
+    this.waitForConnectReady()
+
+    if (!this.isAppInstalled(installAppPayload.installedAppId!!)) {
+      this.installApp(installAppPayload)
+
+      if (enableAfterInstall) {
+        this.enableApp(installAppPayload.installedAppId!!)
+      }
+    }
+
+    return this.ensureAppWebsocket(installAppPayload.installedAppId!!)
+  }
+
   /// Connect to the service
   fun connect() {
+    Log.d(TAG, "connect")
     val intent = Intent()
     intent.setComponent(ComponentName(this.servicePackageName, this.serviceClassName))
     this.activity.bindService(intent, this.mConnection, Context.BIND_ABOVE_CLIENT)
   }
 
+  /// Poll until we are connected to the service, or the timeout has elapsed
+  suspend fun waitForConnectReady(timeoutMs: Long = 5000L, intervalMs: Long = 5L) {
```

Comment:
It takes a few milliseconds between calling `connect` and actually having the IPC binding ready to use.